mantaray/backend.py

```diff
@@ -115,6 +115,11 @@ def __init__(self, server_config: config.ServerConfig, *, verbose: bool):
         self.event_queue: queue.Queue[IrcEvent] = queue.Queue()
         self._thread: threading.Thread | None = None
 
+        # Will contain the capabilities to negotiate with the server
+        self.cap_req: list[str] = []
+        # "CAP LIST" shows capabilities enabled on the client's connection
+        self.cap_list: Set[str] = set()
```

Comment:
These lists should probably be cleared when connecting to the server. Otherwise they can contain whatever was left over from the previous connection, if reconnecting happens in the middle of capability negotiation, for example.